snap/squashfs/squashfs.go

```diff
@@ -171,20 +172,48 @@ func (s *Snap) Size() (size int64, err error) {
 	return st.Size(), nil
 }
 
-// ReadFile returns the content of a single file inside a squashfs snap.
-func (s *Snap) ReadFile(filePath string) (content []byte, err error) {
+func (s *Snap) withUnpackedFile(filePath string, f func(p string) error) error {
 	tmpdir, err := ioutil.TempDir("", "read-file")
 	if err != nil {
-		return nil, err
+		return err
 	}
 	defer os.RemoveAll(tmpdir)
 
 	unpackDir := filepath.Join(tmpdir, "unpack")
 	if output, err := exec.Command("unsquashfs", "-n", "-i", "-d", unpackDir, s.path, filePath).CombinedOutput(); err != nil {
-		return nil, fmt.Errorf("cannot run unsquashfs: %v", osutil.OutputErr(output, err))
+		return fmt.Errorf("cannot run unsquashfs: %v", osutil.OutputErr(output, err))
+	}
+
+	return f(filepath.Join(unpackDir, filePath))
+}
+
+// RandomAccessFile returns an implementation to read at any given location
+// for a single file inside the squashfs snap.
+func (s *Snap) RandomAccessFile(filePath string) (interface {
+	io.ReaderAt
+	io.Closer
+}, error) {
+	var f *os.File
+	err := s.withUnpackedFile(filePath, func(p string) (err error) {
+		f, err = os.Open(p)
+		return
```

Comment:
(nitpick) naked return, iirc we had vet complaining about this

Reply:
it's based on how long the function is, I think here it's a legitimate usage